src/cmds/codegen/codegen.ts

```diff
@@ -128,27 +127,41 @@ export class Codegen {
         }
 
         if (generator === 'typegen') {
+
+          if (!output.typings || output.typings === '') {
+            throw new Error("Please provide output.typings path to use typegen")
+          }
+
           inputSchemaPath = inputSchemaPath || '**/*.ts'
-          const binPath = await getApolloCodegenBin()
+          const binPath = require.resolve('apollo-codegen').replace('index.js', 'cli.js')
           const tmpSchemaPath = getTmpPath()
           fs.writeFileSync(
             tmpSchemaPath,
             JSON.stringify(await introspect(this.project.getSchema())),
           )
           const args = [
             'generate',
-            input,
+            input || '{binding,prisma}/*.ts',
             '--schema',
             tmpSchemaPath,
             '--output',
             output.typings,
             '--target',
             language,
           ]
+
           const child = crossSpawn.sync(binPath, args)
+        
+          if (child.error) {
+            if (child.error.message === `spawnSync apollo-codegen ENOENT`) {
+              throw new Error(`Generator apollo-codegen is not installed.`)
```

Comment:
Can we make this error message actionable? How can I usually fix this problem?

Reply:
Ideally, we will not hit this at all with this new code!